File: app/src/main/java/me/sunzheng/mana/MyFavoritesActivity.java
package me.sunzheng.mana;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatSpinner;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;

import me.sunzheng.mana.home.HomeApiService;
import me.sunzheng.mana.home.mybangumi.FavoritesFragment;
import me.sunzheng.mana.home.mybangumi.MyFavoritesPresenter;
import me.sunzheng.mana.home.mybangumi.respository.DataRespositoryImpl;
import me.sunzheng.mana.utils.App;

/**
 * MyFaviours activity
 * for list the faviours
 * and click any item goto {@link BangumiDetailsActivity}
 */
public class MyFavoritesActivity extends AppCompatActivity {
    FavoritesFragment fragment;
    AppCompatSpinner spinner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my_faviours);
        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        spinner = findViewById(R.id.spinner);
        fragment = FavoritesFragment.newInstance();
        fragment.setPresenter(new MyFavoritesPresenter(fragment, new DataRespositoryImpl(MyFavoritesActivity.this, ((App) getApplicationContext()).getRetrofit().create(HomeApiService.MyBangumi.class))));
        getSupportFragmentManager().beginTransaction().replace(R.id.contentPanel, fragment).commit();
        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                fragment.onFilter(position);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/me/sunzheng/mana/MyFavoritesActivity.kt
package me.sunzheng.mana

import android.content.SharedPreferences
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.preference.PreferenceManager
import dagger.hilt.android.AndroidEntryPoint
import me.sunzheng.mana.databinding.ActivityMyFavioursBinding
import me.sunzheng.mana.home.mybangumi.FavriouteFragmentCompat
import javax.inject.Inject

/**
 * MyFaviours activity
 * for list the faviours
 * and click any item goto [BangumiDetailsActivity]
 */
@AndroidEntryPoint
class MyFavoritesActivity @Inject constructor() : AppCompatActivity() {
    companion object {
        @JvmStatic
        val KEY_POSITION_INT = "${MyFavoritesActivity::class.java.simpleName}_position"
    }

    val binding: ActivityMyFavioursBinding by lazy {
        DataBindingUtil.setContentView(this, R.layout.activity_my_faviours)
    }
    val preference: SharedPreferences by lazy {
        PreferenceManager.getDefaultSharedPreferences(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setSupportActionBar(binding.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        var position = preference.getInt("KEY_POSITION_INT", 0)
        binding.spinner.onItemSelectedListener = object : OnItemSelectedListener {
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View,
                position: Int,
                id: Long
            ) {
                supportFragmentManager.beginTransaction()
                    .replace(R.id.contentPanel, FavriouteFragmentCompat.newInstance(position))
                    .commit()
                preference.edit().putInt("KEY_POSITION_INT", position).commit()
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {}
        }
        binding.spinner.setSelection(position)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
             onBackPressedDispatcher.onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }
}